### DISCUSSION TITLE
使用https反代时建议开启http2，否则日志需要超长时间加载。

### DISCUSSION THREAD

**Discussion by HankunYu** [jxxghp/MoviePilot#679] [讨论]
### 确认

- [X] 我的版本是最新版本，我的版本号与 [version](https://github.com/jxxghp/MoviePilot/releases/latest) 相同。
- [X] 我已经 [issue](https://github.com/jxxghp/MoviePilot/issues) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经 [Telegram频道](https://t.me/moviepilot_channel) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经修改标题，将标题中的 描述 替换为我遇到的问题。

### 当前程序版本

 v1.2.5

### 问题类型

主程序运行问题

### 问题描述

已经按照说明配置了反代设置,但是访问实时日志的时候无限在转刷新
是我设置上有问题么?开了ssl,其他功能一切正常

### 发生问题时系统日志和配置文件

```bash
16         location / {
 17                         proxy_pass http://127.0.0.1:4000;
 18                         proxy_set_header Host $http_host;
 19                         proxy_set_header X-Real-IP $remote_addr;
 20                         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 21                         #proxy_set_header    X-Forwarded-Host $http_host;
 22                         proxy_set_header    X-Forwarded-Proto $scheme;
 23                         proxy_http_version  1.1;
 24                         #proxy_set_header    Host $server_name:$server_port;
 25                         proxy_set_header Upgrade $http_upgrade;
 26                         proxy_set_header Connection keep-alive;
 27
 28                         proxy_set_header Connection "Keep-Alive";
 29                         proxy_set_header Keep-Alive "timeout=60";
 30
 31                         proxy_set_header X-Real-IP $remote_addr;
 32                         proxy_connect_timeout 7d;
 33                         proxy_send_timeout 7d;
 34                         proxy_read_timeout 7d;
 35             root   html;
 36             index  index.html index.htm;
 37         }
```

解决方法：
在server区块下加入
```
http2 on;
ssl_session_cache shared:SSL:10m;
ssl_session_timeout 10m;
ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
ssl_session_tickets on;
```

**Comment by HankunYu** [comment]
找到原因了...开了https就不行

**Comment by ReddotCleaner** [comment]
是不是websocket没开

**Comment by HankunYu** [comment]
> 是不是websocket没开

开了...他要跑很久很久才能建立连接

**Comment by HankunYu** [comment]
连接时长
<img width="558" alt="image" src="https://github.com/jxxghp/MoviePilot/assets/11606969/83dc972c-4ef8-47cd-89f8-259112b41f50">

正常的话应该是这样的（走http
<img width="442" alt="image" src="https://github.com/jxxghp/MoviePilot/assets/11606969/d9a2077a-b951-47b6-84e8-bae2379ad28c">

**Comment by onlyJinx** [comment]
现在好像还是有这个问题，nginx反代的时候日志会加载很久才出来，直接 `ip:port `的话可以秒开

![image](https://github.com/jxxghp/MoviePilot/assets/21324835/23e1150c-2213-4528-9e08-871dfe21e4ac)

**Comment by HankunYu** [comment]
> 现在好像还是有这个问题，nginx反代的时候日志会加载很久才出来，直接 `ip:port `的话可以秒开
> 
> ![image](https://private-user-images.githubusercontent.com/21324835/286196035-23e1150c-2213-4528-9e08-871dfe21e4ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MzI2ODUsIm5iZiI6MTcwOTczMjM4NSwicGF0aCI6Ii8yMTMyNDgzNS8yODYxOTYwMzUtMjNlMTE1MGMtMjIxMy00NTI4LTllMDgtODcxZGZlMjFlNGFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDEzMzk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4MTZmMzI1NmE5NDJmNDNmYmEwZjUwMWQyODJiMDA5ZjdlN2ZhOGJkNzUzZWI5ZmU2YzhkZDI3ZDc1NTliODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Gx7OtJZyDow9LH-za-qM2dCB2AlReGWLtVTtxL930-0)

我找到了解决方法，在nginx配置中添加使用http2，还可以考虑开启ssl session cache。
具体为添加以下代码。
```
http2 on;
ssl_session_cache shared:SSL:10m;
ssl_session_timeout 10m;
ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
ssl_session_tickets on;
```

**Comment by onlyJinx** [comment]
> 我找到了解决方法，在nginx配置中添加使用http2，还可以考虑开启ssl session cache。 具体为添加以下代码。
> 
> ```
> http2 on;
> ssl_session_cache shared:SSL:10m;
> ssl_session_timeout 10m;
> ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
> ssl_session_tickets on;
> ```
试了一下，开了http2后确实可以秒开了，感谢。
